docs/advanced.rst

```diff
@@ -127,6 +127,29 @@ referencing the ``changed_by`` field:
 
 Admin integration requires that you use a ``_history_user.setter`` attribute with your custom ``_history_user`` property (see :ref:`admin_integration`).
 
+Custom ``history_id``
+---------------------
+
+If you want to use something other then an ``AutoField`` for the history
+table primary key you can pass in a ``history_id_field``.  A common use case
+for this would be to use a ``UUIDField``.
+
+For any field provided it will automatically set ``primary_key=True`` and
+``editable=False``.
+
+.. code-block:: python
+
+    import uuid
+    from django.db import models
+    from simple_history.models import HistoricalRecords
+
+    class Poll(models.Model):
+        question = models.CharField(max_length=200)
+        pub_date = models.DateTimeField('date published')
+        history = HistoricalRecords(
+            history_id_field=models.UUIDField(default=uuid.uuid4)
```

Comment:
Might you just want to add a setting that toggles UUIDField vs AutoField? Or do you think it's important to allow any type of field here?

Reply:
We have actually created a model `class AutoUUIDField(models.UUIDField)` which implements the `pre_save` as outlined which is also why I would prefer to keep it as a field instead of a simple flag.

Reply:
What about setting the default field in settings and allowing that to be overwritable? That way, if I want to use UUIDField on every `history_id` without explicitly stating that, I can do so. I.e., if `SIMPLE_HISTORY_HISTORY_ID_DEFAULT_FIELD` is undefined, then use `AutoField` as the default, otherwise use whatever is defined in that setting. Then, on a case-by-case basis, a dev can overwrite the `history_id_field` for a specific model. If you did it with the setting, would you even have the need to overwrite on a case-by-case basis?

Reply:
@rwlogel @rossmechanic I think having a boolean setting in `settings.py` such as `SIMPLE_HISTORY_HISTORY_ID_USE_UUID` as well as allowing overriding the ID field directly in the `HistoricalRecords` constructor is a good trade-off. This would make it easy to use UUID fields across a project by default (a common substitute for `AutoField`) while still allowing for more customized IDs for those that need it.

Reply:
I like this idea. @rwlogel do you mind doing this instead? Solves for the more commonly used issue (using UUIDs instead of sequential IDs) and allows users like yourself to overwrite them with whatever id you want to use, wherever you want to use it

Reply:
Using `SIMPLE_HISTORY_HISTORY_ID_DEFAULT_FIELD` accomplishes the same thing as  `SIMPLE_HISTORY_HISTORY_ID_USE_UUID` with more flexibility.  How to set it up for UUID is documented in the advanced section so it should be very easy for someone to just copy that.

Reply:
But instantiating the field in the settings file prevents us from ever being able to have custom field names for `history_id`. I.e., if in the future, someone is using `history_id` in their model and thus needs to call this historical table's `history_id` something else, this would prevent that, since there's only one instantiation of the UUIDField in settings. Renaming history columns isn't something we support currently, but it's something that people have asked for that I plan to allow in the future to prevent name clashing. If you look at the example below:
```
field = models.CharField(max_length=50)

class TestModel(models.Model):
    test = field

class TestModel1(models.Model):
    test1 = field
```
both `TestModel` and `TestModel1` will have a `CharField` named `test` rather than `TestModel` being named `test` and `TestModel1` being named `test1`. We'll run into the same issue by instantiating in settings. That's why I'd prefer the setting to be a boolean, since most people's use case will be UUIDField (and we can instantiate a new UUIDField each time), and for users who want something other than UUIDField or AutoField, they can overwrite in the `HistoricalRecords` parameters.

Reply:
Good point